R/fetch_mailinglist.R

```diff
@@ -0,0 +1,44 @@
+
+#' Download a mailing list from Qualtrics
+#'
+#' @param mailinglistID String. Unique ID for the mailing list you want to download. Returned as \code{id} by the \link[qualtRics]{all_mailinglists} function.
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' mailinglists <- all_mailinglists()
+#' mailinglist <- fetcH_mailinglist(mailinglists$id[[1]])
+#' }
+
+fetch_mailinglist <- function(mailinglistID){
+
+  assert_base_url()
+  assert_api_key()
+
+  fetch_url <- create_mailinglist_url(Sys.getenv("QUALTRICS_BASE_URL"), mailinglistID)
+
+  elements <- list()
+
+  while(!is.null(fetch_url)){
+
+    res <- qualtrics_api_request("GET", url = fetch_url)
+    elements <- append(elements, res$result$elements)
```

Comment:
`res$result$elements` is a list object here, is it not?  If so, I'm not sure you'd end up with the intended `elements` final list if you have >1 pass through this loop.  See, for example:

```r
test <- list(a = 1:10, b = -1:-10)
append(test, test)
```

Reply:
Ah I understand what you're doing better now!  You're completely right, and my example was way off the mark even before your clarification; I'd forgotten that the data returned from this endpoint is formatted as elements at the case level.  My mistake, and sorry for wasting your time!

(Looks like I even read my own code wrong, too!  Pretty obvious looking back at it I was just starting with purrr and didn't really grasp it well at all.  I'm glad you are coming at this with a better sense.)

Reply:
Related to the subsequent binding, though, in trying to figure out my own code from earlier I stumbled on a possible way to preserve `responseHistory` and `emailHistory` elements as list-columns.  With pipes for ease of reading:
```r
elements %>% 
  purrr::map(
    ~purrr::modify_if(.x, purrr::is_empty, ~NA) %>% 
      purrr::modify_if(is.list, list)
  ) %>% 
  bind_rows %>% 
  tidyr::unnest_wider(col = embeddedData)
```
Not sure how desirable including those would be; it preserves data but the resulting dataframe isn't as easily writeable. 

(Also, question: I'm noticing that pretty much everything is coming back from the endpoint as character-mode already, even my numeric ID codes.  Are you seeing the same?  I didn't upload this specific list I'm looking at so I'm not sure what the default behavior is.)

Reply:
I thought about preserving `responseHistory` and `emailHistory` as list columns, but to your point, the resulting object isn't as clean. That and I felt that response and email history should probably be handled by functions geared towards fetching distribution history (which I'm currently working on).

Overall I think the function should mimic the output you'd see if you download the contact list from Qualtrics directly.

Reply:
And yes the objects I'm seeing returned are of type character.